submitit/core/job_environment.py

```diff
@@ -58,6 +58,10 @@ def activated(self) -> bool:
     def hostname(self) -> str:
         return socket.gethostname()
 
+    @property
```

Comment:
nit: should it be Sequence[str] ?
Also I'd prefer it to return an empty list or [self.hostname] rather than crashing.
Otherwise any user that use hostnames will need to wrap the code with `if cluster == slurm`

Reply:
I will put a Sequence instead of the List.

For the exception, I tend to prefer explicit errors rather than silent errors (meaning getting a result that is not valid), and so in that case raising an exception rather than having an empty list. But I also tend to prefer reasonable defaults to exceptions. So we could settle for [self.hostname]